cosima_cookbook/database.py

```diff
@@ -600,16 +568,60 @@ def build_index(
         directories = [directories]
 
     indexed = 0
-    for directory in directories:
-        indexed += index_experiment(
-            directory, session, client, update, prune, delete, followsymlinks
-        )
+    for directory in [Path(d) for d in directories]:
+        # find all netCDF files in the experiment directory
+        files = find_files(directory, followsymlinks=followsymlinks)
+        expt = find_experiment(session, directory)
+
+        if expt is None:
+            expt = NCExperiment(
+                experiment=str(directory.name), root_dir=str(directory.resolve())
+            )
+        else:
+            if not update:
+                print(
+                    "Not re-indexing experiment: {}\n"
+                    "Pass `update=True` to build_index()".format(directory.name)
+                )
+                continue
```

Comment:
Worth nuking this? Seems to be covered by `force`

Reply:
They are subtly different. `update=False, force=False` will not add *any* entries to an existing experiment. `update=True, force=False` will add only new files to an existing experiment, and `update=True, force=True` will add new files and re-read existing files and add to an experiment.

`update=False, force=True` will ignore the `force` argument, which is borderline buggy. If keeping both should issue a warning at the very least.